tests/pybullet_helpers/test_motion_planning.py

```diff
@@ -0,0 +1,393 @@
+    path = run_motion_planning(robot,
+                               joint_initial,
+                               joint_target,
+                               collision_bodies={table_id, block_id},
+                               seed=seed,
+                               physics_client_id=physics_client_id)
+    assert path is None
+    p.removeBody(block_id, physicsClientId=physics_client_id)
+
+
+def test_move_to_shelf():
+    """Test for Panda robot moving to put a held block into a shelf.
+
+    Notably, the robot must change its gripper orientation from top-down
+    to forward-facing, so motion planning must be in position and
+    orientation.
+
+    Also notably, the held object must be collision-checked like the robot.
+    """
+    utils.reset_config({"pybullet_control_mode": "reset"})
+
+    # Set up scene.
+    x_lb = 1.2
+    x_ub = 1.5
+    y_lb = 0.4
+    y_ub = 1.1
+    pick_z = 0.75
+    default_orn = (0.0, 0.0, 0.0, 1.0)
+    table_pose = (1.35, 0.75, 0.0)
+    table_orientation = (0., 0., 0., 1.)
+    table_height = 0.2
+    shelf_width = (x_ub - x_lb) * 0.4
+    shelf_length = (y_ub - y_lb) * 0.6
+    shelf_base_height = pick_z * 0.8
+    shelf_ceiling_height = pick_z * 0.2
+    shelf_ceiling_thickness = 0.01
+    shelf_pole_girth = 0.01
+    shelf_color = (0.5, 0.3, 0.05, 1.0)
+    shelf_x = x_ub - shelf_width / 2
+    shelf_y = y_lb + shelf_length / 2
+    block_color = (1.0, 0.0, 0.0, 1.0)
+    block_size = 0.05
+    block_x = (x_lb + x_ub) / 2
+    block_y = y_ub - block_size
+    block_z = table_height + block_size / 2
+    offset_z = 0.01
+    obj_mass = 0.5
+    obj_friction = 1.2
+    camera_distance = 0.8
+    camera_yaw = 90.0
+    camera_pitch = -24
+    camera_target = (1.65, 0.75, 0.42)
+    robot_ee_home_orn = (0.7071, 0.7071, 0.0, 0.0)
+    home_pose = Pose((block_x, block_y, block_z + offset_z), robot_ee_home_orn)
+
+    # Target for motion planning.
+    tx = shelf_x
+    ty = shelf_y
+    tz = table_height + shelf_base_height + block_size / 2 + offset_z
+    target_orn = (0.7071, 0.0, 0.7071, 0.0)
+    target_pose = Pose((tx, ty, tz), target_orn)
+
+    if USE_GUI:  # pragma: no cover
+        physics_client_id = p.connect(p.GUI)
+        p.configureDebugVisualizer(p.COV_ENABLE_GUI,
+                                   False,
+                                   physicsClientId=physics_client_id)
+        p.configureDebugVisualizer(p.COV_ENABLE_RGB_BUFFER_PREVIEW,
+                                   False,
+                                   physicsClientId=physics_client_id)
+        p.configureDebugVisualizer(p.COV_ENABLE_DEPTH_BUFFER_PREVIEW,
+                                   False,
+                                   physicsClientId=physics_client_id)
+        p.configureDebugVisualizer(p.COV_ENABLE_SEGMENTATION_MARK_PREVIEW,
+                                   False,
+                                   physicsClientId=physics_client_id)
```

Comment:
Might be worth moving these to some utils as it's used a few times over different modules: https://github.com/Learning-and-Intelligent-Systems/predicators/search?q=p.configureDebugVisualizer